src/overlays/actors/ovl_En_Toryo/z_en_toryo.c

```diff
@@ -29,27 +30,343 @@ const ActorInit En_Toryo_InitVars = {
+    s32 pad;
+    Player* player = PLAYER;
+    s32 ret = 5;
+
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+        case 5:
+            ret = 5;
+            break;
+        case 6:
+            if (func_80106BC8(globalCtx) != 0) {
+                ret = 0;
+            }
+    }
+    return ret;
+}
+
+u32 func_80B20634(EnToryo* this, GlobalContext* globalCtx) {
+    u32 ret;
+
+    if (this->unk_1E0 != 0) {
+        if (this->unk_1E0 == 10) {
+            func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+            if ((gSaveContext.infTable[23] & 0x2)) {
+                ret = 0x606E;
+            } else {
+                ret = 0x606D;
+            }
+        } else {
+            ret = 0x200F;
+        }
+    }
+    //! @bug return value may be unitialized
+    return ret;
+}
+
+s32 func_80B206A0(EnToryo* this, GlobalContext* globalCtx) {
+    s32 textId = Text_GetFaceReaction(globalCtx, 0);
+    s32 ret = textId;
+
+    if (textId == 0) {
+        if ((this->stateFlags & 0x1)) {
+            if ((gSaveContext.eventChkInf[9] & 0xF) == 0xF) {
+                ret = 0x606C;
+            } else if ((gSaveContext.infTable[23] & 0x1)) {
+                ret = 0x606B;
+            } else {
+                ret = 0x606A;
+            }
+        } else if ((this->stateFlags & 0x2)) {
+            if ((gSaveContext.infTable[23] & 0x4)) {
+                ret = 0x5029;
+            } else {
+                ret = 0x5028;
+            }
+        } else {
+            ret = textId;
+            if ((this->stateFlags & 0x4)) {
+                ret = 0x506C;
+            }
+        }
+    }
+    return ret;
+}
+
+void func_80B20768(EnToryo* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 sp32;
+    s16 sp30;
+
+    if (this->unk_1E4 == 3) {
+        func_8002F194(&this->actor, globalCtx);
+        func_8010B720(globalCtx, this->actor.textId);
+        this->unk_1E4 = 1;
+    }
+
+    if (this->unk_1E4 == 1) {
+        this->unk_1E4 = func_80B203D8(this, globalCtx);
+    }
+
+    if (this->unk_1E4 == 5) {
+        this->unk_1E4 = func_80B205CC(this, globalCtx);
+        return;
+    }
+
+    if (this->unk_1E4 == 2) {
+        func_8010B720(globalCtx, this->actor.textId);
+        this->unk_1E4 = 1;
+    }
+
+    if (this->unk_1E4 == 4) {
+        if (Actor_HasParent(&this->actor, globalCtx)) {
+            this->actor.parent = NULL;
+            this->unk_1E4 = 5;
+        } else {
+            func_8002F434(&this->actor, globalCtx, 0x22, 100.0f, 10.0f);
+        }
+        return;
+    }
+
+    if (this->unk_1E4 == 0) {
+        if (func_8002F194(&this->actor, globalCtx) != 0) {
+            this->unk_1E0 = func_8002F368(globalCtx);
+            if (this->unk_1E0 != 0) {
+                player->actor.textId = func_80B20634(this, globalCtx);
+                this->actor.textId = player->actor.textId;
```

Comment:
Does this still match by any chance? It's probably better if it works
```suggestion
                this->actor.textId = player->actor.textId = func_80B20634(this, globalCtx);
```